packages/SwingSet/src/controller/startXSnap.js

```diff
@@ -0,0 +1,96 @@
+import fs from 'fs';
+import path from 'path';
+import { Fail } from '@agoric/assert';
+import { type as osType } from 'os';
+import { xsnap, recordXSnap } from '@agoric/xsnap';
+
+const NETSTRING_MAX_CHUNK_SIZE = 12_000_000;
+
+/**
+ * @param {{ moduleFormat: string, source: string }[]} bundles
+ * @param {{
+ *   snapStore?: SnapStore,
+ *   spawn: typeof import('child_process').spawn
+ *   debug?: boolean,
+ *   traceFile?: string,
+ * }} options
+ */
+export function makeStartXSnap(bundles, options) {
+  // our job is to simply curry some authorities and settings into the
+  // 'startXSnap' function we return
+  const { snapStore, spawn, debug = false, traceFile } = options;
+
+  let serial = 0;
+  const makeTraceFile = traceFile
+    ? () => {
+        const workerTrace = path.resolve(`${traceFile}/${serial}`) + path.sep;
+        serial += 1;
+        return workerTrace;
+      }
+    : undefined;
+
+  /**
+   * @param {string} vatID
+   * @param {string} name
+   * @param {(request: Uint8Array) => Promise<Uint8Array>} handleCommand
+   * @param {boolean} [metered]
+   * @param {boolean} [reload]
+   */
+  async function startXSnap(
+    vatID,
+    name,
+    handleCommand,
+    metered,
+    reload = false,
+  ) {
+    /** @type { import('@agoric/xsnap/src/xsnap').XSnapOptions } */
+    const xsnapOpts = {
+      os: osType(),
+      spawn,
+      stdout: 'inherit',
+      stderr: 'inherit',
+      debug,
+      netstringMaxChunkSize: NETSTRING_MAX_CHUNK_SIZE,
+    };
+
+    let doXSnap = xsnap;
+    if (makeTraceFile) {
+      doXSnap = opts => {
+        const workerTrace = makeTraceFile();
```

Comment:
Likewise here, maybe
```suggestion
        const dirPath = nextDirPath();
```